tests/inputs/TestAssemblyXmlDocument.cpp

```diff
@@ -0,0 +1,94 @@
+//
+
+#include "AssemblyXmlDocument.h"
+#include "Tests.h"
+
+#include <array>
+#include <fstream>
+#include <iostream>
+#include <tuple>
+#include <xml/sax/parser/Parser.h>
+#include <xml/sax/parser/ParserException.h>
+#include <xml/sax/parser/ParserFactory.h>
+
+namespace parser = xml::sax::parser;
+
+testing::Environment* initXmlEnvironment();
+
+testing::Environment* const env = initXmlEnvironment();
+
+TEST(AssemblyXmlDocument, readFile) {
+  using dfl::inputs::AssemblyXmlDocument;
+
+  AssemblyXmlDocument doc;
+  parser::ParserFactory factory;
+  const std::string filepath = "res/assembling.xml";
+
+  auto parser = factory.createParser();
+  std::ifstream in(filepath.c_str());
+  ASSERT_TRUE(in);
+  ASSERT_NO_THROW(parser->parse(in, doc, false));
+
+  const auto& macroConnections = doc.macroConnections();
+  ASSERT_EQ(macroConnections.size(), 2);
+  auto macro = macroConnections[0];
+  ASSERT_EQ(macro.id, "VCSToUMeasurement");
+  ASSERT_EQ(macro.connections.size(), 1);
+  ASSERT_EQ(macro.connections.front().var1, "U_IMPIN");
+  ASSERT_EQ(macro.connections.front().var2, "@NAME@_U");
+  macro = macroConnections[1];
+  ASSERT_EQ(macro.id, "VCSToControlledShunts");
+  ASSERT_EQ(macro.connections.size(), 3);
+  std::array<std::tuple<std::string, std::string>, 3> connect_values = {std::make_tuple("shunt_state_@INDEX@", "@NAME@_state"),
+                                                                        std::make_tuple("shunt_isCapacitor_@INDEX@", "@NAME@_isCapacitor"),
+                                                                        std::make_tuple("shunt_isAvailable_@INDEX@", "@NAME@_isAvailable")};
+  for (unsigned int i = 0; i < macro.connections.size(); ++i) {
+    ASSERT_EQ(macro.connections[i].var1, std::get<0>(connect_values[i]));
+    ASSERT_EQ(macro.connections[i].var2, std::get<1>(connect_values[i]));
+  }
+
+  const auto& singleAssociations = doc.singleAssociations();
+  ASSERT_EQ(singleAssociations.size(), 6);
+  auto singleAssoc = singleAssociations.front();
+  ASSERT_EQ(singleAssoc.id, "MESURE_MODELE_1_B.EPIP4");
+  ASSERT_FALSE(singleAssoc.line);
+  ASSERT_FALSE(singleAssoc.tfo);
+  ASSERT_TRUE(singleAssoc.bus);
+  ASSERT_EQ(singleAssoc.bus->voltageLevel, "B.EPIP6");
+  singleAssoc = singleAssociations[2];
+  ASSERT_EQ(singleAssoc.id, "MESURE_I_BOUTR TR 661");
+  ASSERT_FALSE(singleAssoc.bus);
+  ASSERT_FALSE(singleAssoc.line);
+  ASSERT_TRUE(singleAssoc.tfo);
+  ASSERT_EQ(singleAssoc.tfo->name, "BOUTR TR 661");
+  singleAssoc = singleAssociations[4];
+  ASSERT_EQ(singleAssoc.id, "MESURE_I_ADA_SALON");
+  ASSERT_FALSE(singleAssoc.bus);
+  ASSERT_FALSE(singleAssoc.tfo);
+  ASSERT_TRUE(singleAssoc.line);
+  ASSERT_EQ(singleAssoc.line->name, "RQROUL31S.BLA");
+
+  const auto& multiAssociations = doc.multipleAssociations();
+  ASSERT_EQ(multiAssociations.size(), 2);
+  ASSERT_EQ(multiAssociations.front().id, "SHUNTS_MODELE_1_B.EPIP4");
+  ASSERT_EQ(multiAssociations.front().shunt.voltageLevel, "B.EPIP4");
```

Comment:
What is the coverage of this?

Reply:
Checking that multi associations are read properly

Reply:
100% for assembling, 84% for setting.